packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -261,13 +281,23 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 
 			if (shouldClose) {
 				this.connection.close(options.code, options.reason);
-				await once(this.connection, 'close');
+
+				let outerResolve: () => void;
+				const promise = new Promise<void>((resolve) => {
+					outerResolve = resolve;
+				});
+
+				this.connection.onclose = () => {
+					outerResolve();
+				};
```

Comment:
Wouldn't it be cleaner to do this?
```suggestion
				this.connection.onclose = outerResolve;
```